### PR TITLE
DevOps with GitHub Solutions

### PR DESCRIPTION
**Pull request by Whowong**
Adding the solutions for this WTH.

### CODE REVIEW

File: 031-DevOpsWithGitHub/Coach/Challenge07.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+- Branch protection rules
+    - In your repository, go to "Settings"
+    - Select "Branches" on the left hand side.
+    - Select "Add Rule"
+    - For branch pattern enter the default branch such as main or master
+    - Check "Require a pull request before merging" and select "Require review from Code Owners"
+    - Select "Create"
+- The following is needed to run during the pull request and only for the application path:
+  pull_request:
```
Comment (jordanbean-msft) [inline 031-DevOpsWithGitHub/Coach/Challenge07.md:15]:
```yaml
pull_request:
  branches: [ main ]
  paths: Application/**
```

File: 031-DevOpsWithGitHub/Coach/Challenge07.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    branches: [ main ]
+    paths: Application/**
+- Codeowners
+    - A new file at the root of the repo is needed called "CODEOWNERS", you will need to define the path and the owner.  
+        - /Application/ @whowong
+
```
Comment (jordanbean-msft) [inline 031-DevOpsWithGitHub/Coach/Challenge07.md:21]:
```yaml
- /Application/@whowong
```

**Review by jordanbean-msft — CHANGES_REQUESTED** [review]

File: 031-DevOpsWithGitHub/Coach/Challenge02.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Challenge 2 - Centralize your code with GitHub Repos - Coach's Guide
+
+[< Previous Challenge](./Challenge01.md) - **[Home](README.md)** - [Next Challenge >](./Challenge03.md)
+
+## Notes & Guidance
+
+- Some students create their first repository via github desktop.  Make sure students understand how a distributed source control such as git is different from a centralized source control system such as TFVC. 
+
+- To clone a repository via command line: git clone $URL in the directory you want to copy the repository to.
```
[the remark this answers is not in the document]
Comment (Whowong) [reply 031-DevOpsWithGitHub/Coach/Challenge02.md]: Thanks, thats a good idea.  I just added them in.

File: 031-DevOpsWithGitHub/Coach/Challenge02.md [hunk 4]
```diff
@@ -6,11 +6,11 @@
...
+- To clone a repository via command line: `git clone $URL` in the directory you want to copy the repository to.
   - To get the $URL, go to your repository in github.
   - Select "<> Code" which should be a green button.
   - In the clone section, click on the copy button to get the url.
 - Add the files the coaches provided in the same directory.
-- In the command line enter: "Git add --all" - This will add all of the files you just copied to the folder to be be tracked.
-- Now we need to commit our changes by typing git commit -am "My first commit"
-- Finally we need to push to the remote repository in github by doing "git push -u".  You may see a warning which will provide the full command if this is the first time you are doing this to target the github server.
\ No newline at end of file
+- In the command line enter: `Git add --all` - This will add all of the files you just copied to the folder to be be tracked.
```
[the remark this answers is not in the document]
Comment (Whowong) [reply 031-DevOpsWithGitHub/Coach/Challenge02.md]: Thanks!  Just lowered it.

**Review by jordanbean-msft — APPROVED** [review]

File: 031-DevOpsWithGitHub/Coach/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# What The Hack - IoT Hack of the Century - Coach Guide
```
Comment (jrzyshr) [inline 031-DevOpsWithGitHub/Coach/README.md:1]: I'm pretty sure this isn't the "IoT Hack of the Century" coach guide? ;) 

There's a couple of things in the template that didn't get updated.

File: 031-DevOpsWithGitHub/Coach/README.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+## Introduction
+Welcome to the coach's guide for the IoT Hack of the Century What The Hack. Here you will find links to specific guidance for coaches for each of the challenges.
```
Comment (jrzyshr) [inline 031-DevOpsWithGitHub/Coach/README.md:4]: Replace name of hack.

File: 031-DevOpsWithGitHub/Coach/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+Always refer students to the [What The Hack website](https://aka.ms/wth) for the student guide: [https://aka.ms/wth](https://aka.ms/wth)
+
+**NOTE:** Students should **not** be given a link to the What The Hack repo before or during a hack. The student guide does **NOT** have any links to the Coach's guide or the What The Hack repo on GitHub.  
+
+
+## Azure Requirements
+
+This hack requires students to have access to an Azure subscription where they can create and consume Azure resources. These Azure requirements should be shared with a stakeholder in the organization that will be providing the Azure subscription(s) that will be used by the students.
+
+- Azure resources that will be consumed by a student implementing the hack's challenges
```
Comment (jrzyshr) [inline 031-DevOpsWithGitHub/Coach/README.md:48]: what resources are consumed?

File: 031-DevOpsWithGitHub/Coach/README.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+- Students will need access to a subscription or at minimum a resource group.  If the latter permissions are used, the group will need to ensure a student is either able to create a service principle or one will need to be generated in advance.
```
Comment (jrzyshr) [inline 031-DevOpsWithGitHub/Coach/README.md:49]: If what "latter permission" are used?

File: 031-DevOpsWithGitHub/Coach/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+## Repository Contents
+
+- \`./Coach\`
```
Comment (jrzyshr) [inline 031-DevOpsWithGitHub/Coach/README.md:53]: The template you copied from had the monospace ticks escaped out for the automation script.   You should remove all of the escape characters.

**Review by jrzyshr — CHANGES_REQUESTED** [review]
A couple of minor things to be cleaned up in the Coach Guide template.

Also, /Coach/Solution folder should be /Coach/Solutions  <-- plural 'solutions'

File: 031-DevOpsWithGitHub/Coach/README.md (hunk 5 above)
Comment (Whowong) [reply 031-DevOpsWithGitHub/Coach/README.md]: Ha!  You are right, I have updated

File: 031-DevOpsWithGitHub/Coach/README.md (hunk 7 above)
Comment (Whowong) [reply 031-DevOpsWithGitHub/Coach/README.md]: Updated to list out resources consumed

File: 031-DevOpsWithGitHub/Coach/README.md (hunk 8 above)
Comment (Whowong) [reply 031-DevOpsWithGitHub/Coach/README.md]: Added additional context for clarification

File: 031-DevOpsWithGitHub/Coach/README.md (hunk 9 above)
Comment (Whowong) [reply 031-DevOpsWithGitHub/Coach/README.md]: Updated formatting and removed the escaped ticks

Comment (jrzyshr) [reply 031-DevOpsWithGitHub/Coach/README.md]: Doh!  The backticks belong.  Just the escape characters need to be removed as those were there for the script.  (Hopefully no one will need to do this again once they're all generating the template. :))

Comment (Whowong) [reply 031-DevOpsWithGitHub/Coach/README.md]: Added the ticks back in

**Review by jrzyshr — APPROVED** [review]
Thanks for the new Coach Guide!

**Pull request merged**